src/configs/config.controller.ts

```diff
@@ -0,0 +1,59 @@
+import { ConfigService } from "../adapters/default/config.adapter";
+import {
+  ApiTags,
+  ApiBody,
+  ApiOkResponse,
+  ApiForbiddenResponse,
+  ApiCreatedResponse,
+  ApiBasicAuth,
+} from "@nestjs/swagger";
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Put,
+  Param,
+  UseInterceptors,
+  ClassSerializerInterceptor,
+  SerializeOptions,
+  Req,
+  Query,
+  CacheInterceptor,
+} from "@nestjs/common";
+import { ConfigSearchDto } from "./dto/config-search.dto";
+import { Request } from "@nestjs/common";
+import { ConfigDto } from "./dto/config.dto";
+
+@ApiTags("Config")
+@Controller("config")
+export class ConfigController {
+  constructor(
+    private service: ConfigService
+  ) {}
+
+  @Get(":module/all")
+  @UseInterceptors(ClassSerializerInterceptor)
+  @ApiBasicAuth("access-token")
+  @ApiCreatedResponse({ description: "Config detail" })
+  @ApiForbiddenResponse({ description: "Forbidden" })
+  @SerializeOptions({
+    strategy: "excludeAll",
+  })
+  public async getConfigForTeacher(@Req() request: Request) {
+    return this.service.getConfigForTeacher(request);
+  }
+
+  @Post(":module")
+  @ApiBasicAuth("access-token")
+  @ApiCreatedResponse({ description: "Config has been created successfully." })
+  @ApiBody({ type: ConfigDto })
+  @ApiForbiddenResponse({ description: "Forbidden" })
+  @UseInterceptors(ClassSerializerInterceptor)
+  public async createConfig(
+    @Req() request: Request,
+    @Body() configDto: ConfigDto
```

Comment:
This seems to be saving one config at a time. Is there no way to make it save all the configs in a single API call ?

Reply:
Currently, it is saving one config at a time. If we want to save multiple rows at a time, we will have to take request data and traverse a loop and save it.